src/gui/widgets/multiline_text.cpp

```diff
@@ -338,7 +338,7 @@ void multiline_text::handle_key_down_arrow(SDL_Keymod modifier, bool& handled)
 	size_t offset = get_selection_start();
 	const unsigned line_num = get_line_number(offset);
 
-	if (line_num == get_lines_count()) {
+	if (line_num == get_lines_count()-1) {
```

Comment:
I think this makes the check a few line down redundant? if so you can replace it with
```cpp
offset = std::min(offset - line_start + next_line_start, next_line_end);
```

Reply:
Updated the checks in both `down_arrow` and `up_arrow` handler.